### DISCUSSION TITLE
nested factors in how() and adonis

### DISCUSSION THREAD

**Discussion by carburator** [vegandevs/vegan#637] [Q&A]
Hi,

We have 16S fecal microbiome samples from pigs and would like to examine the effects of age (3 levels), treatment (4 levels), gender (2 levels), and the interactions, on microbiome composition.  We have a total of 321 samples taken from 107 pigs birthed from 14 sows (~6-10 piglets per sow of both genders).  Each pig was sampled at 3 timepoints. All pigs from each sow were given the same treatment, with 3-4 sows per treatment. For a variety of reasons, it was not possible to assign different treatments to pig from the same sow.  We would like to use adonis, and restrict the permutations using how(). In the following, pig, sow, treatment, and gender are treated as factors, and age as an ordered factor.

So far:

```r
data < -read.csv("table_feces.csv", header=TRUE, row.names=1)
meta <- read.csv("meta_feces.csv", header=TRUE)
bc <- vegdist(data, method="bray")   

h <- how(plots = Plots(strata = pig, type = c("free")), Within(type = c("series")))

adonis2(formula = bc ~ age * treat * gender, permutations = h)
``` 

From what I understand, piglets are represented as plots here, and permutations are performed across pigs so we can examine the treatment effect.  A series type permutation is performed within each pig to account for the sampling across time.  But, the sow effect is not accounted for. I believe piglet should be nested within sow.

1. Does this makes sense so far? 
2. How would you include piglet nested within sow using how()?  Or would an alternative approach be best?  On similar posts, I've seen the recommendation to use dbrda() for complex experimental designs.  If an alternative method is best would you mind providing example code?

Thanks very much!

**Comment by carburator** [comment]
Also, I should include one more piece of information.  For the analysis above, I eliminated some data so that the design would be balanced such that permutations could be performed across pigs.  If there is a way to also permute across sows above, I believe I would also balance the data for this.

**Comment by gavinsimpson** [comment]
You can't develop a single permutation design to test all the effects of interest. As far as I can follow, treatment effects don't vary *within* piglet, so any test of this effect will need to permute the piglets while keeping the samples within piglet fixed.

If you care to expand on the specific test you want to do, we can perhaps help further.

Apologies for not replying to this earlier - it just slipped my memory.
